var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -80,7 +81,11 @@ class NetcdfC(AutotoolsPackage):
     variant("zstd", default=True, description="Enable ZStandard compression", when="@4.9.0:")
     variant("optimize", default=True, description="Enable -O2 for a more optimized lib")
     variant("nczarr", default=True, description="Enable zarr storage support", when="@4.8.0:")
-    variant("byterange", default=False, description="Allow byte-range I/O")
```

Comment:
@AlexanderRichert-NOAA The way this is written now should throw an error with earlier versions of netcdf-c, because the variant/build option doesn't exist.

Reply:
Since eventually we'll want this, and it's a feature that exists since 4.7.0, why don't we just throw in a version restriction?